tests/components/kodi/test_binary_sensor.py

```diff
@@ -0,0 +1,78 @@
+"""The tests for Kodi binary sensor platform."""
+import logging
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_component import async_update_entity
+
+from . import init_integration
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@pytest.mark.parametrize(
+    "element, name",
+    [("screensaver", "Screensaver"), ("energy_saving", "Energy saving")],
+)
+async def test_binary_sensor_initial_states(hass: HomeAssistant, element, name):
+    """Test for binary sensor values."""
+    entry = await init_integration(hass)
+
+    state = hass.states.get(f"binary_sensor.{entry.data['name']}_{element}")
+    assert state.state == STATE_OFF
+    assert state.attributes["friendly_name"] == f"{entry.data['name']} {name}"
+
+
+@pytest.mark.parametrize("element", ["screensaver", "energy_saving"])
+async def test_binary_sensor_statechange(hass: HomeAssistant, element):
+    """Test the event."""
+    entry = await init_integration(hass)
+
+    state = hass.states.get(f"binary_sensor.{entry.data['name']}_{element}")
```

Comment:
This was already asserted in the previous test

Reply:
I included it to make sure that the sensor is really off at the beginning before switching it to on.

Reply:
Yeah, but unless there's some randomness to it, it was already tested